### PR TITLE
[mono] Fix warnings with recent clang/gcc

### PR DESCRIPTION
**Pull request by akoeplinger**
- Unused/unset variables in hot_reload.c
- Misleading indentation in aot-compiler.c
- Cast to smaller integer type in method-to-ir.c

### CODE REVIEW

**Review by lambdageek — APPROVED** [review]

File: src/mono/mono/mini/aot-compiler.c [hunk 1]
```diff
@@ -6697,23 +6697,27 @@ emit_method_code (MonoAotCompile *acfg, MonoCompile *cfg)
 		 * - it allows the setting of breakpoints of aot-ed methods.
 		 */
 
-		// Comment out to force dedup to link these symbols and forbid compiling
+		// Enable to force dedup to link these symbols and forbid compiling
 		// in duplicated code. This is an "assert when linking if broken" trick.
-		/*if (mono_aot_can_dedup (method) && (acfg->aot_opts.dedup || acfg->aot_opts.dedup_include))*/
-			/*debug_sym = mono_aot_get_mangled_method_name (method);*/
-		/*else*/
+#if FALSE
```
Comment (lambdageek) [inline src/mono/mono/mini/aot-compiler.c:6702]:
```suggestion
#if 0
```

File: src/mono/mono/mini/aot-compiler.c [hunk 2]
```diff
@@ -6697,23 +6697,27 @@ emit_method_code (MonoAotCompile *acfg, MonoCompile *cfg)
...
 		if (acfg->need_no_dead_strip)
 			fprintf (acfg->fp, "	.no_dead_strip %s\n", debug_sym);
 
-		// Comment out to force dedup to link these symbols and forbid compiling
+		// Enable to force dedup to link these symbols and forbid compiling
 		// in duplicated code. This is an "assert when linking if broken" trick.
-		/*if (mono_aot_can_dedup (method) && (acfg->aot_opts.dedup || acfg->aot_opts.dedup_include))*/
-			/*emit_global_inner (acfg, debug_sym, TRUE);*/
-		/*else*/
+#if FALSE
```
Comment (lambdageek) [inline src/mono/mono/mini/aot-compiler.c:6716]:
```suggestion
#if 0
```

**Review by lambdageek — COMMENTED** [review]
C usually uses `#if 0` not `FALSE`

File: src/mono/mono/mini/aot-compiler.c (hunk 1 above)
Comment (am11) [reply src/mono/mono/mini/aot-compiler.c]: few other instances of this pattern are in these three files:
```
$ git grep -l "#if .*FALSE" :/*.c

src/mono/mono/metadata/class.c
src/mono/mono/mini/mini-generic-sharing.c
src/mono/mono/mini/simd-intrinsics.c
```

**Comment by akoeplinger** [comment]
The `Build Browser wasm Release AllSubsets_Mono` failure is https://github.com/dotnet/runtime/issues/62021

**Pull request merged**